### PR TITLE
web: implement a simple windowing mechanism to improve UI perf

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @jazzdan,

Please review the following commits I made in branch nicks/ch4315/windowing:

ba0a276347a6ce79a1ba4948d385d2aa98db61bf (2020-01-23 20:24:26 -0500)
web: implement a simple windowing mechanism to improve UI perf

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/LogPane.tsx [hunk 1]
```diff
@@ -21,6 +21,15 @@ type LogPaneProps = {
...
 }
 
+type LogPaneState = {
+  // The number of log lines to display
+  renderWindow: number
+}
+
+const renderWindowDefault = 50
+const renderWindowStep = 50
+const blankLogLineHeight = 30
```
Comment (hyu) [inline web/src/LogPane.tsx]: Nit: We could use styled-components for the logLine CSS, and use this const value, to ensure it's accurate. For now, though, maybe a comment to note that it's just a value that tracks the computed height of each log line? (Which is currently based on font size and line height.)

Comment (nicks) [reply web/src/LogPane.tsx]: That wouldn't work, because a log line can have multiple lines of text, so the height isn't fixed

**Comment by nicks** [comment]
re: blank artifacts: ya, that is, in many ways, the whole point of this PR. If you forced the browser to render every row off-screen, on the off-chance that the user might scroll fast enough to see it, then you would necessarily make the overall experience much slower.

**Comment by nicks** [comment]
i tweaked the algorithm a bit to expand the render window a bit faster, which should address some of the blank artifact issues.

File: web/src/LogPane.tsx (hunk 1 above)
Comment (nicks) [reply web/src/LogPane.tsx]: added come comments for clarity

**Pull request merged**